app/concerns/koudoku/subscription.rb

```diff
@@ -35,7 +35,7 @@ def processing!
 
             # update the package level with stripe.
             if sub = customer.subscriptions.first
-              sub.prorate = Koudoku.prorate
+              sub.prorate = Koudoku.prorate || (upgrading? ? Koudoku.prorate_for_upgrade : Koudoku.prorate_for_downgrade)
```

Comment:
would put the least specific at the end like

`(upgrading? ? Koudoku.prorate_for_upgrade : Koudoku.prorate_for_downgrade) || Koudoku.prorate`